### PR TITLE
scripted approach to hand flares, fix #1079

### LINKED ISSUE #1079: M127A1 model flickers
http://youtu.be/7O_IZQl1Ey4

### PR DESCRIPTION
**Pull request by commy2**
This is the only way I know of to fix the flickering flares #1079 

benefits:
- hand flares can bounce like normal grenades
- they have impact sounds
- no flickering
- more options to adjust light

disadvantages:
- it's scripted
- it's not config

### CODE REVIEW

File: addons/grenades/functions/fnc_flare.sqf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// by commy2
```
Comment (jokoho48) [inline addons/grenades/functions/fnc_flare.sqf]: Wrong Docu

File: addons/grenades/CfgAmmo.hpp [hunk 2]
```diff
@@ -28,66 +28,66 @@ class CfgAmmo {
...
-        SmokeShellSoundHit1[] = {"A3\Sounds_F\weapons\smokeshell\smoke_1",1.25893,1,100};
-        SmokeShellSoundHit2[] = {"A3\Sounds_F\weapons\smokeshell\smoke_2",1.25893,1,100};
-        SmokeShellSoundHit3[] = {"A3\Sounds_F\weapons\smokeshell\smoke_3",1.25893,1,100};
-        SmokeShellSoundLoop1[] = {"A3\Sounds_F\weapons\smokeshell\smoke_loop1",0.125893,1,70};
-        SmokeShellSoundLoop2[] = {"A3\Sounds_F\weapons\smokeshell\smoke_loop2",0.125893,1,70};
         timeToLive = 60;
     };
 
     class SmokeShell;
+    class ACE_G_Handflare_White: SmokeShell {
```
Comment (jokoho48) [inline addons/grenades/CfgAmmo.hpp]: this destroy missions

Comment (commy2) [inline addons/grenades/CfgAmmo.hpp]: It doesn't the old ones are still there.
All these smokegrenadeXXX entries didn't do anything anyway. (simualtion = "shotIllumination")

**Comment by SzwedzikPL** [comment]
I like this :+1: I've suggested @KoffeinFlummi scripted flickering fix long time ago. I'm glad that is finally fixed :)

File: addons/grenades/functions/fnc_throwGrenade.sqf [hunk 3]
```diff
@@ -21,22 +21,50 @@
...
 
-if (_unit != ACE_player) exitWith {};
 if (_weapon != "Throw") exitWith {};
 
+// http://feedback.arma3.com/view.php?id=12340
+if (isNull _projectile) then {
+    _projectile = nearestObject [_unit, _ammo];
+};
+
+// handle speial grenades
```
Comment (Glowbal) [inline addons/grenades/functions/fnc_throwGrenade.sqf]: special

**Comment by alganthe** [comment]
No backwards compatibility missing since there were no compatibility needed in the first place.
. :+1:

**Pull request merged**